File: packages/@simpli/cli-scaffold/generator/dynamics/src/model/Template.ts
/**
 * <%-model.name%>
 * @author Simpli© CLI generator
 */
<%_ for (var i in model.resolvedDependencies) { var dependence = model.resolvedDependencies[i] _%>
<%-dependence.build()%>
<%_ } _%>

export default class <%-model.name%> extends <%-model.isResource ? 'Resource' : 'Model'%> {
<%_ if (model.isResource) { _%>
<%-model.buildResource()%>
<%_ } _%>
<%_ for (var i in model.attrs) { var attr = model.attrs[i] _%>
<%-attr.build()%>
<%_ } _%>
<%_ for (var i in model.apis) { var api = model.apis[i] _%>
<%-api.build()%>
<%_ } _%>
<%_ if (model.isResource) { _%>
<%-model.buildScheme()%>
<%-model.buildScheme(true)-%>
<%_ } _%>
}

File: packages/@simpli/cli-server/generator/injected/src/main/model/resource/Template.kt
<%_ var packageAddress = options.serverSetup.packageAddress _%>
<%_ var startCase = options.serverSetup.startCase _%>
package <%-packageAddress%>.model.resource

import <%-packageAddress%>.exception.response.BadRequestException
import <%-packageAddress%>.model.param.DefaultParam
import br.com.simpli.model.LanguageHolder
import br.com.simpli.tools.Validator
import io.swagger.v3.oas.annotations.media.Schema
import java.util.Date
import javax.ws.rs.PathParam

/**
 * Reference model of table <%-table.name%>
<%_ if (table.commentary) { _%>
 * Note: <%-table.commentary%>
 *
<%_ } _%>
 * @author Simpli CLI generator
 */
<%_ if (table.commentary) { _%>
@Schema(description = "<%-table.commentary%>")
<%_ } _%>
class <%-table.modelName%> {
<%-table.buildIdsColumns()-%>
<%-table.buildRelations()-%>
<%-table.buildCommentaryColumns()-%>
<%-table.buildRequiredAndMaxlengthAndNotDescriptionColumns()-%>
<%-table.buildRequiredAndNotMaxlengthAndNotDescriptionColumns()-%>
<%-table.buildMaxlengthAndNotRequiredAndNotDescriptionColumns()-%>
<%-table.buildSimpleColumns()-%>
<%-table.buildPasswordColumns()-%>
<%-table.buildRequiredPathId()-%>
<%-table.buildGettersAndSetters()-%>
    fun validate(lang: LanguageHolder) {
<%-table.buildValidate()-%>
    }
}
